Plugins/RSBot.Map/Views/Main.cs

```diff
@@ -269,8 +274,12 @@ private void PopulateMapAndGrid(Graphics graphics)
                     {
                         foreach (var entry in coses)
                         {
-                            AddGridItem(entry.Name, "", entry.Record.Level, entry.Movement.Source);
-                            DrawPointAt(graphics, entry.Movement.Source, 1);
+                            // Avoid painting vehicles from main player
+                            if(Game.Player.Vehicle?.Id != entry.Id)
```

Comment:
Ah yes vehicle. Sorry. 

But about uniqueId, it will be uniqueId because you are selecting the coses from spawn list. So uniqueId checking are right.